src/audio.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2020  Joshua Minter & Patryk Seregiet
+ *
+ * This file is part of ReplaySorcery.
+ *
+ * ReplaySorcery is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ReplaySorcery is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with ReplaySorcery.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "audio.h"
+#include "util/log.h"
+#include "util/memory.h"
+
+static void rsAudioReadSamples(RSAudio *audio);
+static void rsAudioPrepareEncoder(RSAudio *audio, const RSConfig* config);
+static void rsAudioEncoderPrepareFrame(RSAudioEncoder *audioenc);
+
+extern sig_atomic_t mainRunning;
```

Comment:
Please do not have globals like this. Pass a struct to the thread, have main call an audio function to modify values in regards to that struct. Also `sig_atomic` mainly applies to signal handlers

Reply:
I initially had a structure, but since it only had one member I just changed it to RSAudio pointer. When talking about globals you mean that atomic bool ? I'm not sure if I can get rid of it. You are right about the wrong type though. I just copied the "mainRunning" without thinking it's a type for signals.

Reply:
The whole comment is in regard to the atomic bool. I only have globals when absolutely necessary and even then I definitely do not make it global to the whole program. Just have a structure (potentially the `RSAudio` structure) passed to the thread and change a bool on that by calling an `rsAudio` function.